app/scripts/components/common/map/controls/aoi/custom-aoi-control.tsx

```diff
@@ -76,15 +76,25 @@ function CustomAoI({
   const [selectedState, setSelectedState] = useState('');
   const [presetIds, setPresetIds] = useState([]);
   const [fileUploadedIds, setFileUplaodedIds] = useState([]);
+  const [, forceUpdate] = useState(0);   // @NOTE:  Needed so that this component re-renders to when the draw selection changes from feature to point.
+  const [isAreaSelected, setAreaSelected] = useState<boolean>(false);
+  const [isPointSelected, setPointSelected] = useState<boolean>(false);
 
   const [selectedForEditing, setSelectedForEditing] = useAtom(selectedForEditingAtom);
 
   const { onUpdate, isDrawing, setIsDrawing, features } = useAois();
   const aoiDeleteAll = useSetAtom(aoiDeleteAllAtom);
 
-  // Needed so that this component re-renders to when the draw selection changes
-  // from feature to point.
-  const [, forceUpdate] = useState(0);
+  // @NOTE: map?._drawControl?.getSelected() needs access to mapboxgl draw context store,
+  // but the function gets called before mapboxdraw store is initialized (before being added to map) resulting in an error
+  useEffect(() => {
+    if (!map) return;
+
+    const mbDraw = map?._drawControl;
+    setAreaSelected(mbDraw?.getSelected().features.length);
+    setPointSelected(mbDraw?.getSelectedPoints()?.features.length);
+  }, [map]);
```

Comment:
It looks like the hook isn’t triggered after the initial load, - which makes sense since its only dependency is map.

You can check this by drawing two polygons, and select only one. the message at the bottom should say 'delete area' (not Delete All Areas) like [this page](https://earthdata.nasa.gov/dashboard/exploration?datasets=%5B%5D&taxonomy=%7B%7D&search=&selectedForEditing=true&aois=%5B%22xs%60aJcbujBypqnB%7DraC%3Fnuu%7BC%7Czv_Cgjjc%40%22%2C%2295e0ec%22%2Cfalse%2C%22fczjEeeh%7CAyn%7DnDgcqo%40q%7EzK%60rqwCb%7CbyAfsnoA%22%2C%2269e741%22%2Ctrue%5D) but the message doesn't change. (because the state didnt' change - because the hook was not triggered 💦 )

Reply:
Thanks for catching 🙏🏼. Fixed.